File: laramie-tesoreria-webapi/src/domain/entities/pago-rendicion.ts
export default class PagoRendicion {

    id: number;
	idPagoRendicionLote: number;
	idCuentaPago: number;
	codigoDelegacion: string;
	numeroRecibo: number;
	codigoLugarPago: string;
	importePago: number;
	fechaPago: Date;
	codigoBarras: string;

	constructor(
        id: number = 0,
		idPagoRendicionLote: number = 0,
		idCuentaPago: number = 0,
		codigoDelegacion: string = "",
		numeroRecibo: number = 0,
		codigoLugarPago: string = "",
		importePago: number = 0,
		fechaPago: Date = null,
		codigoBarras: string = ""
	)
	{
        this.id = id;
		this.idPagoRendicionLote = idPagoRendicionLote;
		this.idCuentaPago = idCuentaPago;
		this.codigoDelegacion = codigoDelegacion;
		this.numeroRecibo = numeroRecibo;
		this.codigoLugarPago = codigoLugarPago;
		this.importePago = importePago;
		this.fechaPago = fechaPago;
		this.codigoBarras = codigoBarras;
	}

	setFromObject = (row) =>
	{
        this.id = row.id ?? 0;
		this.idPagoRendicionLote = row.idPagoRendicionLote ?? 0;
		this.idCuentaPago = row.idCuentaPago ?? 0;
		this.codigoDelegacion = row.codigoDelegacion ?? "";
		this.numeroRecibo = row.numeroRecibo ?? 0;
		this.codigoLugarPago = row.codigoLugarPago ?? "";
		this.importePago = row.importePago ?? 0;
		this.fechaPago = row.fechaPago ?? null;
		this.codigoBarras = row.codigoBarras ?? "";
	}

}

File: modules/tesoreria/models/pago_rendicion.py
from sqlalchemy import Column, BigInteger, String, Numeric, DateTime, ForeignKey
from sqlalchemy.orm import relationship

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class PagoRendicion(Base):
    __tablename__ = "tesoreria_pago_rendiciones"

    id = Column(BigInteger, primary_key=True)
    id_pago_rendicion_lote = Column(BigInteger, ForeignKey("tesoreria_pago_rendicion_lotes.id"), nullable=False)
    id_cuenta_pago = Column(BigInteger, nullable=True)
    codigo_delegacion = Column(String(50), nullable=True)
    numero_recibo = Column(String(50), nullable=True)
    codigo_lugar_pago = Column(String(50), nullable=True)
    importe_pago = Column(Numeric(18, 2), nullable=False)
    fecha_pago = Column(DateTime, nullable=True)
    codigo_barras = Column(String(250), nullable=True)

    lote = relationship("PagoRendicionLote", back_populates="rendiciones", lazy="joined")
